elements/modal.js

```diff
@@ -0,0 +1,68 @@
+const widget = require('cache-element/widget')
+const Modal = require('../lib/modal-element')
+const html = require('choo/html')
+
+module.exports = createWidget
+
+function createWidget () {
+  var link = null
+
+  var modal = Modal(html`<div></div>`, function () {
+    link = null
+  })
+
+  return widget({
+    onload: function () {
+      if (link) modal.show(createHtml(link))
+    },
+    onunload: function () {
+      link = null
+      modal.hide()
+    },
+    onupdate: function (el, linkArg) {
+      if (!link && linkArg) {
+        // fresh link
+        link = linkArg
+        modal.show(createHtml(link))
+      } else if (!linkArg) {
+        // link was cleared
+        link = null
+        modal.hide()
+      } else if (linkArg && (link !== linkArg)) {
+        // ohey, we moved to a different link
+        link = linkArg
+        modal.show(createHtml(link))
+      }
+    },
+    render: function (linkArg) {
+      link = linkArg
+      if (link) {
+        modal.show(createHtml(link))
+      } else {
+        modal.hide()
+      }
+
+      return html`<div>${modal}</div>` // this fixes a bug in modal-element
```

Comment:
The issue is in `lib/` - there's no corresponding GH issue yet as I'm not sure what the cause is; you're right I should create one tho; doing that :sparkles: